README.md

```diff
@@ -32,132 +34,201 @@ This table contains all the `WinetProxy` instances that act as entry-points for
 
 - **`bytecode()`**:
   - _Description_:
-    - Returns Witnet RADON script as a CBOR-encoded `bytes`. 
+    - Returns Witnet Data Request as a CBOR-encoded `bytes`. 
+
+- **`codehash()`**:
+  - _Description_:
+    - Returns SHA256 hash of Witnet Data Request as CBOR-encoded bytes.
+
+## **WitnetRequestBoard**
 
+From the point of view of a `UsingWitnet` contract, any given `WitnetRequestBoard` will always support the following interfaces:
 
-## WitnetRequestBoard
+- **`IWitnetRequestBoardEvents`**
+- **`IWitnetRequestBoardRequestor`**
+- **`IWitnetRequestBoardView`**
 
-The `WitnetRequestBoard` implements the following functionality:
+### IWitnetRequestBoardEvents:
 
-- **`estimateGasCost(uint256 _gasPrice)`**:
+- Event **`PostedRequest(uint256 queryId, address from)`**:
   - _Description_: 
-    - Estimates the minimal amount of reward needed to post a Witnet data request into the WRB, for a given gas price.
+    - Emitted when a Witnet Data Request is posted to the WRB.
+  - _Arguments_:
+    - `queryId`: the query id assigned to this new posting.
+    - `from`: the address from which the Witnet Data Request was posted.
+
+- Event **`PostedResult(uint256 queryId, address from)`**:
+  - _Description_: 
+    - Emitted when a Witnet-solved result is reported to the WRB.
+  - _Arguments_:
+    - `queryId`: the id of the query the result refers to.
+    - `from`: the address from which the result was reported.
+
+- Event **`DeletedQuery(uint256 queryId, address from)`**:
+  - _Description_: 
+    - Emitted when all data related to given query is deleted from the WRB.
+  - _Arguments_:
+    - `queryId`: the id of the query that has been deleted.
+    - `from`: the address from which the result was reported.    
+
+### IWitnetRequestBoardRequestor:
+
+- **`postRequest(IWitnetRequest _request)`**:
+  - _Description_: 
+    - Posts a Witnet Data Request into the WRB in the expectation that it will be eventually relayed and resolved 
+  by the Witnet decentralized oracle network, with `msg.value` as reward.
   - _Inputs_:
-    - `_gasPrice`: the request contract address which includes the Witnet RADON script bytecode.
+    - `_request`: the actual `IWitnetRequest` contract address which provided the Witnet Data Request bytecode.
   - _Returns_:
-    - The minimal reward amount. 
+    - *_id*: the unique identifier of the data request.
 
-- **`destroyResult(uint256 _id)`**:
+- **`upgradeReward(uint256 _id)`**:
+  - _Description_: increments the reward of a Witnet data request by 
+  adding more value to it. The new data request reward will be increased by `msg.value`.
+  - _Inputs_:
+    - `_id`: the unique identifier of a previously posted Witnet data request.
+
+- **`deleteQuery(uint256 _id)`**:
   - _Description_:
     - Retrieves the Witnet-solved result (if already available) of a previously posted Witnet request, and removes it from the WRB storage.
```

Comment:
This function removes the entire request, doesn't it? In this comment looks like only the result is deleted